src/screens/LoginScreen.js

```diff
@@ -23,9 +23,9 @@ const {campo_libre, tagline, login_as_guest, login_with_facebook} = login;
 class LoginScreen extends Component {
 
     componentWillMount() {
-        const {navigation: {navigate}} = this.props;
+        const {token, navigation: {navigate}} = this.props;
 
-        this.props.checkAndSetToken({navigate});
+        this.props.checkAndSetToken({token, navigate});
```

Comment:
If props already has a `token` (which it will after a successful rehydration) pass it along. If it's `undefined` then the reducers will pass along the initial map state anyway.